src/Common/src/TypeSystem/IL/Stubs/CalliIntrinsic.cs

```diff
@@ -45,7 +45,8 @@ public static MethodIL EmitIL(MethodDesc target)
 
             var signature = new MethodSignature(template.Flags, 0, returnType, parameters);
 
-            codeStream.Emit(ILOpcode.calli, emitter.NewToken(signature));
+            //codeStream.Emit(ILOpcode.calli, emitter.NewToken(signature));
+            DelegateThunk.EmitTransformedCalli(emitter, codeStream, signature);
```

Comment:
Nit: I know you have said that you did not bother to find a good home for this ... but just moving EmitTransformedCalli from DelegateThunks.cs to CalliIntrinsic.cs would make things a bit nicer.